test/integration/tests/hierarchycontrol.sh

```diff
@@ -0,0 +1,66 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+source helpers.sh
+
+cleanup() {
+
+    shut_down
+}
+trap cleanup EXIT
+
+start_up
+
+tpm2_clear
+
+# ERROR: phEnable may not be SET using this command
+tpm2_hierarchycontrol -c p -a phEnable set 2>&1 1>/dev/null | grep "ERROR"
+
+# EROOR: Only platform hierarchy handle can be specified for SET
+tpm2_hierarchycontrol -c o -a shEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c o -a ehEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c o -a phEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c o -a phEnableNV set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a shEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a ehEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a phEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a phEnableNV set 2>&1 1>/dev/null | grep "ERROR"
+
+# ERROR: Permanent handle lockout not supported by this command
+tpm2_hierarchycontrol -c l -a shEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a ehEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a phEnable set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a phEnableNV set 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a shEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a ehEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a phEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c l -a phEnableNV clear 2>&1 1>/dev/null | grep "ERROR"
+
+# ERROR: Only platform and its authorization can be specified for CLEAR
+tpm2_hierarchycontrol -c o -a ehEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c o -a phEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c o -a phEnableNV clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a shEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a phEnable clear 2>&1 1>/dev/null | grep "ERROR"
+tpm2_hierarchycontrol -c e -a phEnableNV clear 2>&1 1>/dev/null | grep "ERROR"
```

Comment:
shouldn't these error cases return non-zero to the shell and cause the error trap to run? I don't see the trap handler being cleared for negative testing.

Reply:
Oh it's probably because grep is succeeding, just do (off top of my head so likely close but wrong):

```sh
# functionality tests here
trap - err 
# negative tests here
```

Reply:
I have modified test method.
Would you help to check it again?